### PR TITLE
Layout variations

### PR DESCRIPTION
**Pull request by torotil**

### CODE REVIEW

File: campaignion_layout/campaignion_layout.module [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    'theme callback' => 'campaignion_layout_get_theme_for_entity',
+    'theme arguments' => ['node', 1],
+  ];
+  $no_callback = [
+    'theme callback' => 'campaignion_layout_no_custom_theme',
+    'theme arguments' => [],
+  ];
+  $set_theme = [
+    'view' => TRUE,
+    'share' => TRUE,
```
Comment (torotil) [inline campaignion_layout/campaignion_layout.module:32]: `node/%/continue` is missing.

File: campaignion_layout/campaignion_layout.field.inc [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    '#default_value' => $item['theme'],
+    '#id' => $theme_id,
+  ];
+  $element['layout'] = [
+    '#type' => 'value',
+    '#value' => $item['layout'],
+  ];
+  $fields = [];
+  foreach ($themes as $theme_name => $theme) {
+    $options = ['' => t('- None -')];
```
Comment (torotil) [inline campaignion_layout/campaignion_layout.field.inc]: Let’s call this “Default layout”.

File: campaignion_layout/campaignion_layout.module [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+ */
+function campaignion_layout_no_custom_theme() {
+  return NULL;
+}
+
+/**
+ * Menu theme callback: Set the theme based on layout variation fields.
+ *
+ * @see campaignion_layout_menu_alter()
+ * @see campaignion_layout_menu_get_item_alter()
```
Comment (torotil) [inline campaignion_layout/campaignion_layout.module]: This comment is outdated.

File: campaignion_layout/campaignion_layout.module [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+ */
+function campaignion_layout_get_theme_for_entity($entity_type, $entity) {
+  if ($item = _campaignion_layout_get_field_item($entity_type, $entity)) {
+    return $item['theme'];
```
Comment (torotil) [inline campaignion_layout/campaignion_layout.module]: Check if theme has layout variations enabled.

**Review by torotil — COMMENTED** [review]
Comments from review with Katharina.

**Review by katzmo — APPROVED** [review]

**Review by katzmo — APPROVED** [review]

**Pull request merged**